pkgs/development/python-modules/pycdio/default.nix

```diff
@@ -29,23 +32,33 @@ buildPythonPackage rec {
     patchShebangs .
   '';
 
+  build-system = [ setuptools ];
+
   nativeBuildInputs = [
-    nose
-    pkgs.pkg-config
-    pkgs.swig
+    pkg-config
+    swig
   ];
-  buildInputs = [
-    setuptools
-    pkgs.libcdio
-  ] ++ lib.optional stdenv.isDarwin pkgs.libiconv;
+
+  dependencies = [ libcdio ] ++ lib.optional stdenv.isDarwin libiconv;
+
+  nativeCheckInputs = [ pytestCheckHook ];
 
   # Run tests using nosetests but first need to install the binaries
   # to the root source directory where they can be found.
-  checkPhase = ''
+  preCheck = ''
     ./setup.py install_lib -d .
```

Comment:
Running `setup.py` directly is deprecated so we should figure out a way to avoid this.